dm/dm/master/openapi.go

```diff
@@ -732,36 +802,37 @@ func (s *Server) DMAPIOperateTableStructure(ctx echo.Context, taskName string, s
 		opReq.Flush = *req.Flush
 	}
 	workerReq := workerrpc.Request{Type: workerrpc.CmdOperateSchema, OperateSchema: opReq}
-	newCtx := ctx.Request().Context()
+	newCtx := c.Request.Context()
 	resp, err := worker.SendRequest(newCtx, &workerReq, s.cfg.RPCTimeout)
 	if err != nil {
-		return err
+		_ = c.Error(err)
+		return
 	}
 	if !resp.OperateSchema.Result {
-		return terror.ErrOpenAPICommonError.New(resp.OperateSchema.Msg)
+		_ = c.Error(terror.ErrOpenAPICommonError.New(resp.OperateSchema.Msg))
+		return
 	}
-	return nil
 }
 
-func terrorHTTPErrorHandler(err error, c echo.Context) {
-	var code int
-	var msg string
-	if tErr, ok := err.(*terror.Error); ok {
-		code = int(tErr.Code())
-		msg = tErr.Error()
-	} else {
-		msg = err.Error()
-	}
-	if sendErr := sendHTTPErrorResp(c, code, msg); sendErr != nil {
-		c.Logger().Error(sendErr)
+func terrorHTTPErrorHandler() gin.HandlerFunc {
```

Comment:
I am not confident in the implementation of this middleware... wait for another reviewe with better suggestion

Reply:
I think it is good.